src/clients/bridges/BaseAdapter.ts

```diff
@@ -113,88 +111,63 @@ export abstract class BaseAdapter {
-
-    assert(l1Tokens.length === associatedL1Bridges.length, "Token and bridge arrays are not the same length");
+  async checkAndSendTokenApprovals(address: string, l1Tokens: string[], l1Bridges: string[]): Promise<void> {
+    this.log("Checking and sending token approvals", { l1Tokens, l1Bridges });
+    assert(l1Tokens.length === l1Bridges.length, "Token and bridge arrays are not the same length");
 
-    const tokensToApprove: { l1Token: Contract; targetContract: string }[] = [];
     const l1TokenContracts = l1Tokens.map(
       (l1Token) => new Contract(l1Token, ERC20.abi, this.getSigner(this.hubChainId))
     );
-    const redis = await getRedisCache(this.logger);
+
     const allowances = await Promise.all(
-      l1TokenContracts.map(async (l1TokenContract, index) => {
-        // If there is not both a l1TokenContract and associatedL1Bridges[index] then return a number that wont send
-        // an approval transaction. For example not every chain has a bridge contract for every token. In this case
-        // we clearly dont want to send any approval transactions.
-        if (l1TokenContract && associatedL1Bridges[index]) {
-          // Check if we've cached already that this allowance is high enough. Returning null means we won't
-          // send an allowance approval transaction.
-          if (redis) {
-            const key = await this.getAllowanceCacheKey(l1TokenContract.address, associatedL1Bridges[index]);
-            const result = await redis.get<string>(key);
-            if (result !== null) {
-              const savedAllowance = toBN(result);
-              if (savedAllowance.gte(toBN(MAX_SAFE_ALLOWANCE))) {
-                return null;
-              }
-            }
-          }
-          return l1TokenContract.allowance(address, associatedL1Bridges[index]);
-        } else {
-          return null;
+      l1TokenContracts.map(async (l1Token, idx) => {
+        const l1Bridge = l1Bridges[idx];
+
+        // If there is not both a l1TokenContract and l1Bridge then return a number that wont send an approval
+        // transaction. For example not every chain has a bridge contract for every token. In this case we clearly
+        // don't want to send any approval transactions.
+        if (!l1Token && l1Bridge) {
+          return undefined;
+        }
+
+        // Check if we've cached already that this allowance is high enough, else fallback to an RPC query.
+        let allowance = await getTokenAllowanceFromCache(l1Token.address, address, l1Bridge);
+        if (allowance) {
+          return allowance;
         }
+
+        // If the onchain allowance is > MAX_SAFE_ALLOWANCE, cache it for next time.
+        allowance = await l1Token.allowance(address, l1Bridge);
+        if (allowance.gte(MAX_SAFE_ALLOWANCE)) {
+          await setTokenAllowanceInCache(l1Token.address, address, l1Bridge, allowance);
+        }
+
+        return allowance;
```

Comment:
We may be able to reduce the diff if we wrapped this 
```suggestion
        if(!allowance) {
          // If the onchain allowance is > MAX_SAFE_ALLOWANCE, cache it for next time.
          allowance = await l1Token.allowance(address, l1Bridge);
          if (allowance.gte(MAX_SAFE_ALLOWANCE)) {
            await setTokenAllowanceInCache(l1Token.address, address, l1Bridge, allowance);
          }
        }
        return allowance;
```

Reply:
Good call - implemented: https://github.com/across-protocol/relayer/pull/1620/commits/9ee231677a4d97ede40ed3f83f2aa60ad99dede3